cocoa/src/toga_cocoa/window.py

```diff
@@ -45,10 +46,40 @@ def windowShouldClose_(self, notification) -> bool:
 
     @objc_method
     def windowDidResize_(self, notification) -> None:
-        if self.interface.content:
+        if getattr(self, "interface", None) is not None and self.interface.content:
             # Set the window to the new size
             self.interface.content.refresh()
 
+    @objc_method
+    def windowDidMiniaturize_(self, notification) -> None:
+        self.impl._requested_state_applied = True
+        self.impl._process_pending_state()
+
+    @objc_method
+    def windowDidDeminiaturize_(self, notification) -> None:
+        self.impl._requested_state_applied = True
+        self.impl._process_pending_state()
+
+    @objc_method
+    def windowDidEnterFullScreen_(self, notification) -> None:
+        self.performSelector_withObject_afterDelay_(SEL("enteredFullScreen:"), None, 0)
+
+    @objc_method
+    def enteredFullScreen_(self, sender) -> None:
+        # Doing the following directly in `windowDidEnterFullScreen_` will result in error:
+        # ````2024-08-09 15:46:39.050 python[2646:37395] not in fullscreen state````
+        # and any subsequent window state calls to the OS will not work or will be glitchy.
+        self.impl._in_full_screen = True
+        self.impl._requested_state_applied = True
+        self.impl._process_pending_state()
+
+    @objc_method
+    def windowDidExitFullScreen_(self, notification) -> None:
+        if self.impl:  # pragma: no branch
```

Comment:
As noted with interface - how is this possible? How can the delegate be invoked before the impl has been set?

Reply:
I don't think it's getting the events before the interface has been set, rather it is getting them after the references to `self.interface` and `self.impl` are lost i.e., after the window close call to the OS and before the OS actually closes the window.

I am not sure but I think we need to detach the window delegate before closing the window, so that any window delegate events are not transmitted after we call close on the window.